### PR TITLE
Support undo for a file review

### PR DESCRIPTION
**Pull request by madwort**
* UI buttons could be better, but we can fix that in post 
* fixes https://github.com/opensafely-core/airlock/issues/284
Screenshot 2024-05-24 at 17 08 16

### CODE REVIEW

File: local_db/data_access.py [hunk 1]
```diff
@@ -322,6 +322,19 @@ def reject_file(
...
 
+    def reset_review_file(
+        self, request_id: str, relpath: UrlPath, username: str, audit: AuditEvent
+    ):
+        with transaction.atomic():
+            request_file = RequestFileMetadata.objects.get(
+                request_id=request_id, relpath=relpath
+            )
+
+            review = FileReview.objects.get(file=request_file, reviewer=username)
```
Comment (rebkwok) [inline local_db/data_access.py:333]: Should this catch the exception if the review doesn't exist? (It's reasonably easy to trigger if you have 2 airlock tabs open )

**Review by rebkwok — COMMENTED** [review]
One question, and a failing test because of the recent change to `get_request_file`; otherwise it generally lgtm, although the buttons are getting a bit crazy. Would it make more sense not to have a separate button for the reset review, but have re-clicking approve/reject for an already approved/rejected file do the resetting (could also the tooltip make the tooltip say "File already approved; click to reset"?

File: local_db/data_access.py (hunk 1 above)
Comment (madwort) [reply local_db/data_access.py:333]: yep, have added handler & tests for this.

**Comment by madwort** [comment]
Yeah, I agree that we've got too many buttons right now, but I don't think we should aggregate the buttons right now. I think these buttons are currently the only way that output checkers can see whether they'vereviewed a file or not, and so overloading the buttons in that way would make that a lot more difficult. If we had review status in the file menu on the left, I think it might be ok to overload them?

I think there's a few other ideas for improving the UI - in the original ticket Simon suggested adding another type of widget, another option might be to copy Github & have a "review button" that produces a drop-down with three options, or something else.

What do you think about merging this as-is and then creating another ticket to look at the UI, probably involving Tom O tbh?

**Comment by madwort** [comment]
created an issue to improve the UI https://github.com/opensafely-core/airlock/issues/357

**Pull request merged**